### DISCUSSION TITLE
BiLSTM implementation question

### DISCUSSION THREAD

**Discussion by Matsumotorise** [brianhie/viral-mutation#31] [Q&A]
Hello, 

I was wondering if the following would be an implementation bug or not.

```
for _ in range(n_hidden - 1):
            lstm = LSTM(hidden_dim, return_sequences=True)
            x_pre = lstm(x_pre)
            x_post = lstm(x_post)
        lstm = LSTM(hidden_dim)
        x_pre = lstm(x_pre)
        x_post = lstm(x_post)
```

Does applying the same LSTM to x_pre and x_post mean they share the same weights? I think Keras's Bidirectional wrapper makes the weights distinct in the forwards and backwards operations.  Would this be to reduce model size?

Thank you!

**Comment by brianhie** [comment]
Good question! This is by design, the weights are shared between the forward and backward direction following https://arxiv.org/abs/1902.08661

**Reply by Matsumotorise** [reply]
Thanks!
